jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/MySQLFlexibleServer/MinTLSVersionRule.java

```diff
@@ -0,0 +1,127 @@
+                List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+                Annotation annotation = null;
+                annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                annotation.put(PacmanSdkConstants.DESCRIPTION,
+                        "TLS version for MySQL Flexible server is NOT 1.2");
+                annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                annotation.put(PacmanRuleConstants.CATEGORY, category);
+                issue.put(PacmanRuleConstants.VIOLATION_REASON,
+                        ruleParam.get(PacmanRuleConstants.RULE_ID) + " Violation Found!");
+                issueList.add(issue);
+                annotation.put(PacmanRuleConstants.ISSUE_DETAILS, issueList.toString());
+                logger.debug("TLS version for MySQL Flexible server is NOT 1.2 completed with FAILURE isValid flag {} : ",
+                        isValid);
+                return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                        annotation);
+            }
+        }
+
+        logger.debug("TLS version for MySQL Flexible server is 1.2 with Success isValid flag {}", isValid);
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+
+    }
+
+    private boolean validateTLSVersion(String esUrl, Map<String, Object> mustFilter) throws Exception {
+        logger.info("Validating the resource data from elastic search. ES URL:{}, FilterMap : {}", esUrl, mustFilter);
+        boolean validationResult = true;
+        JsonParser parser = new JsonParser();
+        JsonObject resultJson = RulesElasticSearchRepositoryUtil.getQueryDetailsFromES(esUrl, mustFilter,
+                new HashMap<>(),
+                HashMultimap.create(), null, 0, new HashMap<>(), null, null);
+        logger.debug("Data fetched from elastic search. Response JSON: {}", resultJson);
+
+        if (resultJson != null && resultJson.has(PacmanRuleConstants.HITS)) {
+            String hitsString = resultJson.get(PacmanRuleConstants.HITS).toString();
+            logger.debug("hit content in result json: {}", hitsString);
+            JsonObject hitsJson = (JsonObject) parser.parse(hitsString);
+            JsonArray hitsJsonArray = hitsJson.getAsJsonObject().get(PacmanRuleConstants.HITS).getAsJsonArray();
+            if (hitsJsonArray.size() > 0) {
+                JsonObject jsonDataItem = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                        .get(PacmanRuleConstants.SOURCE);
+                logger.debug("Validating the data item: {}", jsonDataItem);
+                String tlsVersion = jsonDataItem.getAsJsonObject()
+                        .get(PacmanRuleConstants.AZURE_TLSVersion_MySQLFlex).getAsString();
+                if (tlsVersion.equals("TLSv1.2")) {
```

Comment:
Read the required compliant version from rule params

Reply:
taken from rule params